pom.xml

```diff
@@ -338,9 +338,30 @@
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.8.1</version>
           <configuration>
+            <compilerId>javac-with-errorprone</compilerId>
+            <forceJavacCompilerUse>true</forceJavacCompilerUse>
             <source>1.8</source>
             <target>1.8</target>
+            <compilerArgs combine.children="override">
+              <compilerArg>-Xep:MixedMutabilityReturnType:OFF</compilerArg>
+              <compilerArg>-Xep:ImmutableEnumChecker:OFF</compilerArg>
```

Comment:
https://errorprone.info/bugpattern/ImmutableEnumChecker

Alternatively we could annotate every type that's referenced from an enum with `@Immutable`.

Reply:
Do we have many violations?

In general, we do should be careful to only use effectively immutable values in enum, so I don't dislike the idea of the checker at least reminding us to double-check ourselves. So, unless this triggers all the time (but that would feel surprising), I'd have a slight preference for using `@SuppressWarnings("ImmutableEnumChecker")` with a short sentence to remind why it's ok. 

Though in general, I kind of like the use of `@Immutable` (though it does often require use of ``@SuppressWarnings("Immutable")` because the checker is only so good and we have to be ok with that; this mean it maybe require its own discussion and documentation if we want to go there), so I'm up for that as well.